client/resource_group/controller/controller.go

```diff
@@ -254,11 +260,21 @@ func (c *ResourceGroupsController) cleanUpResourceGroup(ctx context.Context) err
 	for _, group := range groups {
 		latestGroups[group.GetName()] = struct{}{}
 	}
-	// TODO: maybe we should also clean up those resource groups that have not been used for a long time.
 	c.groupsController.Range(func(key, value any) bool {
 		resourceGroupName := key.(string)
 		if _, ok := latestGroups[resourceGroupName]; !ok {
 			c.groupsController.Delete(key)
+			return true
+		}
+
+		gc := value.(*groupCostController)
+		// Check stale resource group
+		if equalRU(gc.run.lastRequestConsumption, gc.run.consumption) {
```

Comment:
I think we should use gc.mu.consumption. Consider this scenario: the resource group has some requests, but the runState has not been updated, resulting in the cleanup. Then the resource group will be recreated, but the previously applied RU token will be lost, which may cause the QPS to decrease.